### ISSUE TITLE
Crash when formatting heredoc

### ISSUE THREAD

**Issue by lpil** [ruby-formatter/rufo#48]
Example code that produces the crash:
```ruby
      execute <<~SQL
      SQL
```
```
You've found a bug!
It happened while trying to format the file db/migrate/20170209120024_split_destination_masters_and_files.rb
Please report it to https://github.com/ruby-formatter/rufo/issues with code that triggers it

/home/louis/.asdf/installs/ruby/2.3.0/lib/ruby/2.3.0/ripper/lexer.rb:70:in `block in on_heredoc_dedent': undefined meth
od `event' for #<Array:0x000000021b77c0> (NoMethodError)
        from /home/louis/.asdf/installs/ruby/2.3.0/lib/ruby/2.3.0/ripper/lexer.rb:69:in `each'
        from /home/louis/.asdf/installs/ruby/2.3.0/lib/ruby/2.3.0/ripper/lexer.rb:69:in `on_heredoc_dedent'
        from /home/louis/.asdf/installs/ruby/2.3.0/lib/ruby/2.3.0/ripper/lexer.rb:61:in `parse'
        from /home/louis/.asdf/installs/ruby/2.3.0/lib/ruby/2.3.0/ripper/lexer.rb:61:in `parse'
        from /home/louis/.asdf/installs/ruby/2.3.0/lib/ruby/2.3.0/ripper/lexer.rb:55:in `lex'
        from /home/louis/.asdf/installs/ruby/2.3.0/lib/ruby/2.3.0/ripper/lexer.rb:44:in `lex'
        from /home/louis/.asdf/installs/ruby/2.3.0/lib/ruby/gems/2.3.0/gems/rufo-0.1.0/lib/rufo/formatter.rb:14:in `ini
tialize'
        from /home/louis/.asdf/installs/ruby/2.3.0/lib/ruby/gems/2.3.0/gems/rufo-0.1.0/lib/rufo/command.rb:115:in `new'
        from /home/louis/.asdf/installs/ruby/2.3.0/lib/ruby/gems/2.3.0/gems/rufo-0.1.0/lib/rufo/command.rb:115:in `form
at'
        from /home/louis/.asdf/installs/ruby/2.3.0/lib/ruby/gems/2.3.0/gems/rufo-0.1.0/lib/rufo/command.rb:85:in `forma
t_file'
        from /home/louis/.asdf/installs/ruby/2.3.0/lib/ruby/gems/2.3.0/gems/rufo-0.1.0/lib/rufo/command.rb:68:in `block
 in format_args'
        from /home/louis/.asdf/installs/ruby/2.3.0/lib/ruby/gems/2.3.0/gems/rufo-0.1.0/lib/rufo/command.rb:67:in `each'
        from /home/louis/.asdf/installs/ruby/2.3.0/lib/ruby/gems/2.3.0/gems/rufo-0.1.0/lib/rufo/command.rb:67:in `forma
t_args'
        from /home/louis/.asdf/installs/ruby/2.3.0/lib/ruby/gems/2.3.0/gems/rufo-0.1.0/lib/rufo/command.rb:25:in `run'
        from /home/louis/.asdf/installs/ruby/2.3.0/lib/ruby/gems/2.3.0/gems/rufo-0.1.0/lib/rufo/command.rb:12:in `run'
        from /home/louis/.asdf/installs/ruby/2.3.0/lib/ruby/gems/2.3.0/gems/rufo-0.1.0/exe/rufo:4:in `<top (required)>'
        from /home/louis/.asdf/installs/ruby/2.3.0/bin/rufo:23:in `load'
        from /home/louis/.asdf/installs/ruby/2.3.0/bin/rufo:23:in `<main>'
```

**Comment by mjago** [comment]
@lpil There's a bug in Ruby in 2.3.0 (Ripper). A fix is backported to 2.3.5 and 2.4.2 so you will need to upgrade Ruby to one of those to use squiggly heredoc

**Comment by lpil** [comment]
I see. More motivation for us to get off this older Ruby version. Thanks :)
